src/collections/_documentation/workflow/integrations/integration-platform/index.md

```diff
@@ -3,166 +3,573 @@ title: Integration Platform
+
+## API Token(s)
+
+Sentry's Integration Platform uses API Tokens, which are a similar concept to Access Tokens.
+
+### Public
+
+#### 1. Token Exchange
+
+Upon the initial installation, you'll need the grant code given to you in either the installation webhook request or the redirect URL, in addition to your integration's client ID and client Secret.
+
+```python
+    url = 'https://sentry.io/api/0/sentry-app-installations/{}/authorizations/'
     url = url.format(install_id)
-    
+
     payload = {
         'grant_type': 'authorization_code',
         'code': code,
         'client_id': 'your-client-id',
         'client_secret': 'your-client-secret',
     }
-    
-    resp = requests.post(url, json=payload)
-    data = resp.json()
-    
-    access_token = data['token']
-    refresh_token = data['refreshToken']
-    # ... Securely store the install_id, access_token and refresh_token in DB ...
-    
-    return redirect('http://sentry.io/settings/')
 ```
 
-Next, implement the Webhook URL endpoint, `/webhook`, which receives requests based on your resource subscriptions. In this case that includes when an issue is created. In this method, you'll use the Sentry API to check if the issue belongs to a project called Backend, and if it does, you'll forward the issue to Pushover.
+Tokens expire after eight hours, so you'll need to refresh your tokens accordingly. 
 
 ```python
-@app.route('/webhook', methods=['POST'])
-def webhook():
-    data = json.loads(request.data)
-    
-    if data['action'] != 'created':
-        return
-    
-    issue_id = data['data']['issue']['id']
-    install_id = data['installation']['uuid']
-    
-    issue_details = get_sentry_issue(install_id, issue_id)
-    
-    if issue_details['project']['name'] != 'Backend':
-        return
-        
-    event = data['data']['event']
-    
+    url = 'http://sentry.io/api/0/sentry-app-installations/{}/authorizations/'
+    url = url.format(install_id)
+
+    refresh_token = retrieve_refresh_token_from_db(install_id)
+
     payload = {
-        'user': 'pushover-user-key',
-        'token': 'pushover-api-token',
-        'message': event['message'][:1024],
-        'title': event['message'][:250],
-        'url': event['url'],
-        'url_title': 'Issue Details',
-        'priority': 0,
+        'grant_type': 'refresh_token',
+        'refresh_token': refresh_token,
+        'client_id': 'your-client-id',
+        'client_secret': 'your-client-secret',
     }
-    requests.post('https://api.pushover.net/1/messages.json', data=payload)
+```
+
+The data you can expect back for both the initial grant code exchange and subsequent token refreshes is as follows:
+
+```python
+    {
+        'token': '<example_token_value',
+        'refreshToken': '<example_refresh_token_value>',
+        # all the other things im forgetting plz add thnx bai
```

Comment:
``` json 
{
    "id": "38",
    "token": "ec48bf98637d44c294ead7566513686237e74ab67a074c64b3aaca2d93dbb8f1",
    "refreshToken": "c866f154a65841638d44ee26364409b0a1a67bd642bd46e7a476f34f810712d6",
    "dateCreated": "2019-08-07T20:25:09.870Z",
    "expiresAt": null,
    "state": null,
    "application": null
}
```

Reply:
@mnoble @scefali not sure if we should put in same values for expiresAt and state since those are sometimes there?